ocw/lib/EC2.py

```diff
@@ -216,6 +218,121 @@ def cleanup_all(self):
         self.cleanup_images()
         self.cleanup_snapshots()
         self.cleanup_volumes()
+        self.cleanup_uploader_vpcs()
+
+    def delete_vpc(self, region, vpcId):
+        if PCWConfig.getBoolean('cleanup/vpc_only_notify', self._namespace):
+            return False
+        try:
+            vpc = self.ec2_resource(region).Vpc(vpcId)
+            for subnet in vpc.subnets.all():
+                if len(list(subnet.instances.all())):
+                    self.log_warn('{} has associated instance(s) so can not be deleted', vpcId)
+                    return False
+            self.log_info('{} has no associated instances. Initializing cleanup of it', vpcId)
+
+            self.delete_internet_gw(vpc)
+            self.delete_routing_tables(vpc)
+            self.delete_vpc_endpoints(region, vpcId)
+            self.delete_security_groups(vpc)
+            self.delete_vpc_peering_connections(region, vpcId)
+            self.delete_network_acls(vpc)
+            self.delete_vpc_subnets(vpc)
+            if self.dry_run:
+                self.log_info('Deletion of VPC skipped due to dry_run mode')
+            else:
+                # finally, delete the vpc
+                self.ec2_resource(region).meta.client.delete_vpc(VpcId=vpcId)
+        except Exception as e:
+            self.log_err("{} on VPC {} deletion. {}", type(e).__name__, vpc, traceback.format_exc())
+            send_mail('{} on VPC deletion in [{}]'.format(type(e).__name__, self._namespace), traceback.format_exc())
+            return False
+        return True
+
+    def delete_vpc_subnets(self, vpc):
+        for subnet in vpc.subnets.all():
+            for interface in subnet.network_interfaces.all():
+                if self.dry_run:
+                    self.log_info('Deletion of {} skipped due to dry_run mode', interface)
+                else:
+                    self.log_info('Deleting {}', interface)
+                    interface.delete()
```

Comment:
This seems to be repeated a lot so I suggest to extract a function or method that evaluates the dry-run parameter

Reply:
fully agree with you but since this behavior is already used everywhere in project , and this PR is already pretty huge and complex I would prefer to solve this in separate PR . I created an issue to not forget about it https://github.com/SUSE/pcw/issues/107 .